src/include/sof/math/trig.h

```diff
@@ -137,17 +144,189 @@ static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
+ * Q2.30 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 15
+ * Error (max = 0.000000027939677), THD+N  = -157.454534077921551 (dBc)
+ */
+static inline int32_t asin_fixed_32b(int32_t saturated_unary_minus)
+{
+	int32_t th_rad_fxp;
+
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic_asin(saturated_unary_minus,
+						   CORDIC_31B_TABLE_SIZE,
+						   cordic_ilookup);
+	} else {
+		th_rad_fxp = -is_scalar_cordic_asin(-saturated_unary_minus,
+						    CORDIC_31B_TABLE_SIZE,
+						    cordic_ilookup);
+	}
+	return th_rad_fxp; /* Q2.30 */
+}
+
+/**
+ * CORDIC-based approximation of inverse cosine
+ * inverse cosine of cdc_acos_th based on a CORDIC approximation
+ * acos(X) inverse cosine angle values in radian produces using the DCORDIC
+ * (Double CORDIC) algorithm.
+ * Q2.30 cdc_acos_th , input value range [-1 to 1]
+ * Q3.29 th_rad_fxp, output value range [3.14159265346825 to 0]
+ * LUT size set type 31
+ * Error (max = 0.000000026077032), THD+N  = -157.948952635422842 (dBc)
+ */
+static inline int32_t acosine_fixed_32b(int32_t saturated_unary_minus)
+{
+	int32_t th_rad_fxp;
+
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic_acos(saturated_unary_minus,
+						   CORDIC_31B_TABLE_SIZE,
+						   cordic_ilookup);
+	} else {
+		th_rad_fxp =
+		PI_MUL2_Q4_28 - is_scalar_cordic_acos(-saturated_unary_minus,
+						      CORDIC_31B_TABLE_SIZE,
+						      cordic_ilookup);
+	}
+	return th_rad_fxp; /* Q3.29 */
+}
+
+/**
+ * CORDIC-based approximation of inverse sine
+ * inverse sine of cdc_asin_th based on a CORDIC approximation.
+ * Inverse sine angle values in rad
+ * Q2.30 cdc_asin_th, value in between range of [-1 to 1]
+ * Q1.15 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 31
+ * number of iteration 15
+ * Error (max = 0.000059800222516), THD+N  = -89.824282520774048 (dBc)
+ */
+static inline int32_t asin_fixed_16b(int32_t saturated_unary_minus)
+{
+	int32_t th_rad_fxp;
+
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic_asin(saturated_unary_minus,
+						   CORDIC_16B_ITABLE_SIZE,
+						   cordic_ilookup);
+	} else {
+		th_rad_fxp = -is_scalar_cordic_asin(-saturated_unary_minus,
+						    CORDIC_16B_ITABLE_SIZE,
+						    cordic_ilookup);
+	}
+	/*convert Q2.30 to Q2.14 format*/
+	return sat_int16(Q_SHIFT_RND(th_rad_fxp, 30, 14));
+}
+
+/**
+ * CORDIC-based approximation of inverse cosine
+ * inverse cosine of cdc_acos_th based on a CORDIC approximation
+ * acos(X) inverse cosine angle values in radian produces using the DCORDIC
+ * (Double CORDIC) algorithm.
+ * Q2.30 cdc_acos_th , input value range [-1 to 1]
+ * Q1.15 th_rad_fxp, output value range [3.14159265346825 to 0]
+ * LUT size set type 31
+ * number of iteration 15
+ * Error (max = 0.000059799232976), THD+N  = -89.824298401466635 (dBc)
+ */
+static inline int32_t acosine_fixed_16b(int32_t saturated_unary_minus)
```

Comment:
Same comment as asin_fixed_16b above.
nit: curious why naming "acosine" rather than "acos" since you named "sin", "cos", "asin"

Reply:
Sure, I will rename  acosine to acos